app/scripts/inpage.js

```diff
@@ -121,3 +121,69 @@ function restoreContextAfterImports () {
     console.warn('MetaMask - global.define could not be overwritten.')
   }
 }
+
+/* ------ HANDLE INJECTING PLUGINS ------- */
+
+var plugins = {};
+
+try {
+var rand = function(){ return Math.ceil(Math.random()*10000) }
+// the following should be dynamic, NOT hard coded,
+// based on a UI a user can use to add directories or URLs.
+// We DO NOT want to have to add dependencies to MetaMask for plugins,
+// but for now, I did, just to get the gulp build system working properly.
+plugins.GUN = require('gun/gun.min.js');
+plugins.SEA = require('gun/sea.js');
```

Comment:
For actual inclusion, we would not want full libraries injected into every page, this affects load time of even the smallest iframes. Especially since you're already stubbing out these methods and sending them to the background, we probably would want to import a more "interface" library here, just to save on page size and load time.

Reply:
Very much agree.

If they are already in the background, I don't think they are needed in foreground (or whatever you call it), just the stubs. How do I make sure it stays in the background?